libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -123,6 +123,35 @@ static uint8_t * encodeString( uint8_t * pDestination,
     return pBuffer;
 }
 
+static int32_t recvExact( MQTTTransportRecvFunc_t recvFunc,
+                          MQTTNetworkContext_t networkContext,
+                          void * pBuffer,
+                          size_t bytesToRecv )
+{
+    uint32_t * pIndex = pBuffer;
```

Comment:
Shouldn't it be ```uint8_t *```? Otherwise it will increment in steps of 4.

Reply:
That's right.  Fixed.